interop/xds/server/server.go

```diff
@@ -78,8 +93,76 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}
+
+	for _, headerVal := range getRPCBehaviorMetadata(ctx) {
+		if strings.HasPrefix(headerVal, sleepPfx) {
+			sleep, err := strconv.Atoi(headerVal[len(sleepPfx):])
+			if err != nil {
+				return response, status.Errorf(codes.InvalidArgument, "invalid format for rpc-behavior header: %v", headerVal)
+			}
+
+			time.Sleep(time.Duration(sleep) * time.Second)
+
+		} else if strings.HasPrefix(headerVal, keepOpenVal) {
+			return nil, nil
```

Comment:
Returning `nil, nil` will terminate the RPC. I don't think that is what we want. The spec says: `If the value matches keep-open, the server should never respond to the request and behavior matching ends.`

So, we need to hang here. Something like `<-ctx.Done()` would be more appropriate.